src/IotTelemetrySimulator/SimulatedDevice.cs

```diff
@@ -35,15 +36,21 @@ async Task RunnerAsync(RunnerStats stats, CancellationToken cancellationToken)
                 stats.IncrementDeviceConnected();
 
                 // Delay first event by a random amount to avoid bursts
-                await Task.Delay(this.random.Next(this.interval), cancellationToken);
+                int crtInterval = this.interval[0];
+
+                await Task.Delay(this.random.Next(crtInterval), cancellationToken);
 
                 var stopwatch = new Stopwatch();
                 stopwatch.Start();
+                ulong totalIntervalTime = 0;
                 for (var i = 0L; !cancellationToken.IsCancellationRequested && (this.config.MessageCount <= 0 || i < this.config.MessageCount); i++)
                 {
                     await this.sender.SendMessageAsync(stats, cancellationToken);
 
-                    var millisecondsDelay = Math.Max(0, this.interval * i - stopwatch.ElapsedMilliseconds);
+                    crtInterval = this.interval[i % this.interval.Length];
+                    totalIntervalTime = totalIntervalTime + (ulong)crtInterval;
+
+                    var millisecondsDelay = Math.Max(0, totalIntervalTime - (ulong)stopwatch.ElapsedMilliseconds);
```

Comment:
I am starting to think that trying to adjust the delay using total time won't work well when the simulator is running for weeks or months.

Eventually the totalIntervalTime will cycle back to zero and there won't be any delays. We could drop this adjustment for now, create an issue and revisit the idea later. Or implement a simpler adjustment that only accounts for delays since last message sent.

What do you think?

Reply:
I'll implement a simpler adjustment based on the delay caused by the sending of the last message.

Reply:
I have implemented an adjustment based on the last message and I have tried various techniques, e.g. resetting the stopwatch every time, not reseting the watch and just keeping the counter and I have seen that at bigger numbers (hundreds of messages) we are introducing a total delay if we compensate only for the last message. 

I suggest that we keep compensating on the total duration and reset the counter to 0 every 1000 messages to avoid overflow. What do you think?